rust/datafusion/src/physical_plan/udaf.rs

```diff
@@ -64,6 +64,12 @@ impl Debug for AggregateUDF {
     }
 }
 
+impl PartialEq for AggregateUDF {
```

Comment:
I am wondering, is this the `PartialEq` implementation we always want, or just for this purpose? Otherwise it might be better to create a normal function outside the PartialEq for it? Or do we need it now for a map/set somewhere?

Reply:
I wasn't sure if it was a suitable implementation, and Jorge's raised a good reason why it's not. See thread below. 

The changes in this pull request depend heavily on determining equivalency of two expressions. For example, `rebase_expr()` and `find_exprs_in_expr()` use  `Vec::contains()` (which requires the contained items implement the `PartialEq` trait).